torch/csrc/jit/pybind_utils.h

```diff
@@ -77,6 +78,69 @@ inline TypedIValue toDictKeyIValue(py::handle key) {
   }
 }
 
+inline bool isNamedTupleClass(py::object obj) {
+  auto tuple_type = reinterpret_cast<PyObject*>(&PyTuple_Type);
+  return PyObject_IsSubclass(obj.ptr(), tuple_type) &&
+      py::hasattr(obj, "_fields");
+}
+
+inline TypePtr getOrCreateNamedTupleType(
+  py::object obj,
+  const c10::QualifiedName& qualName,
+  c10::optional<std::vector<TypePtr>> element_types,
```

Comment:
I don't understand where these can be optional.

Reply:
this is because we want to share the named tuple creation with tracing and scripting. 

* tracing: we will infer the inner types of the container type according to inputs, and `element_types` are the inner types that the tracer inferred. 

* scripting: we couldn't always infer the inner types as tracing since we might not have input from python during scripting. What we did here is we use the type from annotations `annotations.ann_to_type` of the named tuple class.